### ISSUE TITLE
[ESP32S3] Using external 32KHz oscillator blocks esp_restart() from working (IDFGH-8522)

### ISSUE THREAD

**Issue by KonssnoK** [espressif/esp-idf#9976]
### Answers checklist.

- [X] I have read the documentation [ESP-IDF Programming Guide](https://docs.espressif.com/projects/esp-idf/en/latest/) and the issue is not addressed there.
- [X] I have updated my IDF branch (master or release) to the latest version and checked that the issue is present there.
- [X] I have searched the issue tracker for a similar issue and not found a similar issue.

### IDF version.

v4.4.2-381-gc97755c9ae5

### Operating System used.

Windows

### How did you build your project?

VS Code IDE

### If you are using Windows, please specify command line type.

PowerShell

### Development Kit.

Custom ESP32S3 board

### Power Supply used.

External 5V

### What is the expected behavior?

esp_restart() should reboot the board

### What is the actual behavior?

esp_restart() halts the CPU.

### Steps to reproduce.

This happens only if we select the 32KHz oscillator as oscillator source

![image](https://user-images.githubusercontent.com/10288926/195809840-53720b6a-9753-4604-9f5c-2097e2b776b3.png)

Using internal oscillator doesn't create the issue

### More Information.

This issue was first reported in 
https://github.com/espressif/esp-idf/issues/9887

but apparently is caused by a different scenario.

**Comment by KonssnoK** [comment]
we've been trying to replicate with a small dedicated setup, but apparently we are not able to..
There seems to be a concatenation of settings that trigger the issue..

**Comment by KonssnoK** [comment]
It seems that the issue is related to our HW design not being compliant with the esp32s3 doc!
We will close this issue for the time being, and reopen in case we still can't go on.

To add more context:
- We use an external oscillator, which is connected to GPIO15.
- we use GPIO16 as an output GPIO
- In the documentation it's written that the oscillator should be connected to GPIO16 and GPIO15 should have a 1nF cap to ground. Also it's written that GPIO15 cannot be used as GPIO.

Ref
https://docs.espressif.com/projects/esp-idf/en/v4.4.2/esp32s3/api-reference/system/system_time.html

When we started using the chip this doc was probably non-existing and we relied on esp32-s3_hardware_design_guidelines_en.pdf !
